packages/@lwc/rollup-plugin/src/__tests__/compilerConfig/fixtures/dynamicImportConfig/dynamicImportConfig.js

```diff
@@ -0,0 +1,9 @@
+import { LightningElement } from 'lwc';
+
+export default class extends LightningElement {
+    ctor;
+
+    async connectedCallback() {
+        this.ctor = await import('test');
```

Comment:
is this valid? I thought the ctor must be a valid constructor, not a promise of a constructor.

Reply:
Ah yes, this should not be valid, however, the validation on the constructor happens at [runtime](https://github.com/salesforce/lwc/pull/3337/files#diff-a284e53a17a2b48cc32e6bf3b121b3acb849a57df886d1cb92a2a46df5aebb7bR585-R589) when the vnodes are created.

The reason this doesn't throw an error is that this test only validates the output of the `@lwc/rollup-plugin` when the `dynamicImportConfig` compiler option is supplied.

I'll update this unit test to the correct usage.

Reply:
+1

Reply:
Updated!